addons/godot-firebase/database/reference.gd

```diff
@@ -69,9 +69,12 @@ func set_listener(listener_ref : Node) -> void:
         _listener = listener_ref
         add_child(_listener)
         _listener.connect("new_sse_event", self, "on_new_sse_event")
-        var base_url = _get_list_url().trim_suffix(_separator)
+        var base_url = _get_list_url(false).trim_suffix(_separator)
         var extended_url = _separator + _db_path + _get_remaining_path(false)
-        _listener.connect_to_host(base_url, extended_url)
+        var port = -1
+        if Firebase.emulating:
+            port = int(_config.emulators.ports.realtimeDatabase)
+        _listener.connect_to_host(base_url, extended_url, port)
```

Comment:
-1 is the default value
The port must be passed as an additional parameter when using `connect_to_host`